### ISSUE TITLE
typescript: index through "class" declarations [phab:D1356]

### ISSUE THREAD

**Issue by kythe** [google/kythe#1751]
_Created by **evmar** at 2017-02-21 22:59:13_: 

Handle "class", member variables, and methods, mostly by
resuing the codepaths for functions and variables.

Add a test with a simple class in it.

**Comment by kythe** [comment]
_**zarko** commented inline (context lost during migration) at 2017-02-21 23:30:49_:

```
class.ts: Could not verify all goals. The furthest we reached was:
  class.ts:4:5-4:31 Base defines/binding Base
```

**Comment by kythe** [comment]
_**zarko** required changes at 2017-02-21 23:30:49_

**Comment by kythe** [comment]
_**evmar** commented inline (context lost during migration) at 2017-02-22 00:00:39_:

I believe this is #1755 .
If you "yarn build" then "yarn test" (which the above change fixes), does this pass?

**Comment by kythe** [comment]
_**zarko** commented inline (context lost during migration) at 2017-02-22 00:13:20_:

`//-` and below.

**Comment by kythe** [comment]
_**zarko** commented inline (context lost during migration) at 2017-02-22 00:13:20_:

It does; I was actually expecting a different bug (see the comment below) and was surprised by the one that #1755 proved invalid.

**Comment by kythe** [comment]
_**evmar** wrote at 2017-02-22 01:46:22_:

  - Merge remote-tracking branch 'origin/master' into typescript-class
  - Merge remote-tracking branch 'origin/master' into typescript-class
  - ok

**Comment by kythe** [comment]
_**evmar** wrote at 2017-02-22 01:46:56_:

Good catch, thanks!  Fixed now.  (And I typo'd "method" instead of "member" below, which the fixed test now covers.)

**Comment by kythe** [comment]
_**zarko** accepted this revision at 2017-02-22 19:12:04_

**Comment by kythe** [comment]
_**zarko** closed via commit b00e194ad6127e4f41ff38cd891f02601398bc42 at 2017-02-22 19:12:27_
